actors/builtin/miner/miner_state_test.go

```diff
@@ -398,6 +398,155 @@ func TestPostSubmissionsBitfield(t *testing.T) {
 	})
 }
 
+func TestVesting_AddLockedFunds_Table(t *testing.T) {
+	vestStart := abi.ChainEpoch(10)
+	vestSum := int64(100)
+
+	testcase := []struct {
+		desc   string
+		vspec  *miner.VestSpec
+		vepocs []int64
+	}{
+		{
+			desc: "vest funds in a single epoch",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   1,
+				StepDuration: 1,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 100, 0},
+		},
+		{
+			desc: "vest funds with period=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   2,
+				StepDuration: 1,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 50, 50, 0},
+		},
+		{
+			desc: "vest funds with period=2 quantization=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   2,
+				StepDuration: 1,
+				Quantization: 2,
+			},
+			vepocs: []int64{0, 0, 0, 100, 0},
+		},
+		{desc: "vest funds with period=3",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   3,
+				StepDuration: 1,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 33, 33, 34, 0},
+		},
+		{
+			desc: "vest funds with period=3 quantization=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   3,
+				StepDuration: 1,
+				Quantization: 2,
+			},
+			vepocs: []int64{0, 0, 0, 66, 0, 34, 0},
+		},
+		{desc: "vest funds with period=2 step=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   2,
+				StepDuration: 2,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 0, 100, 0},
+		},
+		{
+			desc: "vest funds with period=5 step=2",
+			vspec: &miner.VestSpec{
+				InitialDelay: 0,
+				VestPeriod:   5,
+				StepDuration: 2,
+				Quantization: 1,
+			},
+			vepocs: []int64{0, 0, 0, 40, 0, 40, 0, 20, 0},
```

Comment:
It would be interesting to see this one also with an initial delay of 1 (which I expect to offset the whole thing by 1)

Reply:
added thes test below this one.